openquake/commonlib/calculators/event_based.py

```diff
@@ -0,0 +1,305 @@
+    imts = map(from_string, oq.imtls)
+    trunc_level = getattr(oq, 'truncation_level', None)
+    correl_model = readinput.get_correl_model(oq)
+
+    result = AccumDict({(trt_id, str(gsim)): ([], AccumDict())
+                        for gsim in gsims})
+    for rupture, group in itertools.groupby(
+            ses_ruptures, operator.attrgetter('rupture')):
+        sesruptures = list(group)
+        indices = sesruptures[0].site_indices
+        r_sites = (sitecol if indices is None else
+                   site.FilteredSiteCollection(indices, sitecol))
+
+        computer = calc.gmf.GmfComputer(
+            rupture, r_sites, imts, gsims, trunc_level, correl_model)
+        for sr in sesruptures:
+            for gsim_str, gmvs in computer.compute(sr.seed):
+                gmf_by_imt = AccumDict(gmvs)
+                gmf_by_imt.tag = sr.tag
+                gmf_by_imt.r_sites = r_sites
+                result[trt_id, gsim_str][0].append(gmf_by_imt)
+    if getattr(oq, 'hazard_curves_from_gmfs', None):
+        duration = oq.investigation_time * oq.ses_per_logic_tree_path
+        for gsim in gsims:
+            gmfs, curves = result[trt_id, str(gsim)]
+            curves.update(to_haz_curves(
+                sitecol.sids, gmfs, oq.imtls, oq.investigation_time, duration))
+    return result
+
+
+def to_haz_curves(sids, gmfs, imtls, investigation_time, duration):
+    """
+    :param sids: IDs of the given sites
+    :param gmfs: a list of gmf keyed by IMT
+    :param imtls: ordered dictionary {IMT: intensity measure levels}
+    :param investigation_time: investigation time
+    :param duration: investigation_time * number of Stochastic Event Sets
+    """
+    curves = {}
+    for imt in imtls:
+        data = collections.defaultdict(list)
+        for gmf in gmfs:
+            for sid, gmv in zip(gmf.r_sites.sids, gmf[imt]):
+                data[sid].append(gmv)
+        curves[imt] = numpy.array([
+            gmvs_to_haz_curve(data.get(sid, []),
+                              imtls[imt], investigation_time, duration)
+            for sid in sids])
+    return curves
+
+
+@base.calculators.add('event_based')
+class EventBasedCalculator(base.calculators['classical']):
+    """
+    Event based PSHA calculator generating the ruptures only
+    """
+    hazard_calculator = 'event_based_rupture'
+    core_func = compute_gmfs_and_curves
+
+    def pre_execute(self):
+        """
+        Read the precomputed ruptures (or compute them on the fly) and
+        prepare some empty files in the export directory to store the gmfs
+        (if any). If there were pre-existing files, they will be erased.
+        """
+        haz_out = base.get_hazard(self)
+        self.sitecol = haz_out['sitecol']
+        self.rlzs_assoc = haz_out['rlzs_assoc']
+        self.sesruptures = sorted(sum(haz_out['result'].itervalues(), []),
+                                  key=operator.attrgetter('tag'))
+
+        # touch output files
```

Comment:
Since it was already better explained in the docstring, I suggest to remove this comment or change it